src/tests/BaGet.Tests/ServiceCollectionExtensionsTest.cs

```diff
@@ -0,0 +1,115 @@
+        }
+
+        [Fact]
+        public void AskServiceProviderForWellConfiguredSqliteContext()
+        {
+            //Create a IConfiguration with a minimal "Database" object.
+            Dictionary<string, string> initialData = new Dictionary<string, string>();
+            initialData.Add(DatabaseTypeKey, DatabaseType.Sqlite.ToString());
+            initialData.Add(ConnectionStringKey, "blabla");
+
+            IConfiguration configuration = new ConfigurationBuilder().AddInMemoryCollection(initialData).Build();
+
+            ServiceProvider provider = new ServiceCollection()
+                .Configure<BaGetOptions>(configuration)
+                .AddBaGetContext() //Method Under Test
+                .BuildServiceProvider();
+
+            Assert.NotNull(provider.GetRequiredService<SqliteContext>());
+        }
+
+        [Fact]
+        public void AskServiceProviderForSqliteContextWithMissingConnectionString()
+        {
+            //Create a IConfiguration with a minimal "Database" object.
+            Dictionary<string, string> initialData = new Dictionary<string, string>();
+            initialData.Add(DatabaseTypeKey, DatabaseType.Sqlite.ToString());
+            initialData.Add(ConnectionStringKey, string.Empty); //empty equals missing!
+
+            IConfiguration configuration = new ConfigurationBuilder().AddInMemoryCollection(initialData).Build();
+
+            ServiceProvider provider = new ServiceCollection()
+                .Configure<BaGetOptions>(configuration)
+                .AddBaGetContext() //Method Under Test
+                .BuildServiceProvider();
+
+            InvalidOperationException expected = Assert.Throws<InvalidOperationException>(
+                           () => provider.GetRequiredService<SqliteContext>().Database
+                       );
+
+            Assert.Contains(nameof(DatabaseOptions.ConnectionString), expected.Message);
+        }
+
+        [Theory]
+        [InlineData("<invalid>")]
+        [InlineData("")]
+        [InlineData(" ")]
+        public void AskServiceProviderForInvalidDatabaseType(string databaseType)
+        {
+            //Create a IConfiguration with a minimal "Database" object.
+            Dictionary<string, string> initialData = new Dictionary<string, string>();
+            initialData.Add(DatabaseTypeKey, databaseType);
+
+            IConfiguration configuration = new ConfigurationBuilder().AddInMemoryCollection(initialData).Build();
+
+            ServiceProvider provider = new ServiceCollection()
+                .Configure<BaGetOptions>(configuration)
+                .AddBaGetContext() //Method Under Test
+                .BuildServiceProvider();
+
+            var expected = Assert.Throws<InvalidOperationException>(
+                           () => provider.GetRequiredService<IContext>().Database
+                       );
+            //currently the DatabaseType is implemented as Enum => configuration System uses deserialization and we get a 
```

Comment:
>//c [](start = 12, length = 3)

Please add capitalization for commends and spaces between the `//` and the comment